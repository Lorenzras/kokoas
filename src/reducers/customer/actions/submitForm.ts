import { CustomerForm } from './../../../types/forms';
import validateFormState from './validateFormState';
import addTransactCustomers from './addTransactCustomers';

type SubmitForm = (state: CustomerForm) => CustomerForm;

/**
 * Validate then submit the form if no errors
 *
 * @param state
 * @returns
 * @deprecated No flexibility to update state based on fetch status.
 */
const submitForm: SubmitForm = (state) => {
  /** Toggle isSubmitted, then validate each field */

  const validatedForm = validateFormState(state);

  if (!validatedForm.hasError) {
    console.log('no errors');
    addTransactCustomers(state);
  }

  return validatedForm;
};

export default submitForm;