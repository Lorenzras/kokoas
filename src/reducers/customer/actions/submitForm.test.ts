
import 'regenerator-runtime/runtime';
import { custFormStateTestData } from '../../../helpers/test/testData';


import { addCustomersByFormState } from './submitForm';

describe('Submit', ()=> {
  test('is successful', async ()=> {

    await addCustomersByFormState(custFormStateTestData())
      .then((resp) => {
        console.log(resp, 'ERES');
        expect(resp).toMatchSnapshot({ ok: true });
      });


  });
});