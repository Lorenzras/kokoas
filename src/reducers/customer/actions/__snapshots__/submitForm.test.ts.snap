// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Submit is successful 1`] = `
Object {
  "ok": true,
  "result": Object {
    "ids": Array [
      "60",
    ],
    "records": Array [
      Object {
        "id": "60",
        "revision": "1",
      },
    ],
    "revisions": Array [
      "1",
    ],
  },
}
`;
